java/jdbc/ConnectionSamples/ADBQuickStart.java

```diff
@@ -0,0 +1,149 @@
+  final static String DB_USER = "admin";
+  // For security reasons, you will need to provide database password through console.  
+  static String DB_PASSWORD ;
+  final static String CONN_FACTORY_CLASS_NAME="oracle.jdbc.pool.OracleDataSource";
+
+  /*
+   * The sample demonstrates UCP as client side connection pool.
+   */
+  public static void main(String args[]) throws Exception {
+    // For security purposes, you must enter the password through the console 
+    try {
+      Scanner scanner = new Scanner(System.in);
+      System.out.print("Enter the password for Autonomous Database: ");
+      DB_PASSWORD = scanner.nextLine();
+    }
+    catch (Exception e) {
+       System.out.println("ADBQuickStart - Exception occurred : " + e.getMessage());
+    }
+    
+    // Get the PoolDataSource for UCP
+    PoolDataSource pds = PoolDataSourceFactory.getPoolDataSource();
+
+    // Set the connection factory first before all other properties
+    pds.setConnectionFactoryClassName(CONN_FACTORY_CLASS_NAME);
+    pds.setURL(DB_URL);
+    pds.setUser(DB_USER);
+    pds.setPassword(DB_PASSWORD);
+    pds.setConnectionPoolName("JDBC_UCP_POOL");
+
+    // Default is 0. Set the initial number of connections to be created
+    // when UCP is started.
+    pds.setInitialPoolSize(5);
+
+    // Default is 0. Set the minimum number of connections
+    // that is maintained by UCP at runtime.
+    pds.setMinPoolSize(5);
+
+    // Default is Integer.MAX_VALUE (2147483647). Set the maximum number of
+    // connections allowed on the connection pool.
+    pds.setMaxPoolSize(20);
+
+    // Get the database connection from UCP.
+    try (Connection conn = pds.getConnection()) {
+      System.out.println("Available connections after checkout: "
+          + pds.getAvailableConnectionsCount());
+      System.out.println("Borrowed connections after checkout: "
+          + pds.getBorrowedConnectionsCount());
+      // Perform a database operation
+      doSQLWork(conn);
+    }
+    
+    System.out.println("Available connections after checkin: "
+        + pds.getAvailableConnectionsCount());
+    System.out.println("Borrowed connections after checkin: "
+        + pds.getBorrowedConnectionsCount());
+  }  
+
+  /*
+   * Selects 20 rows from the SH (Sales History) Schema that is the accessible to all 
+   * the database users of autonomous database. 
+   */
+  public static void doSQLWork(Connection conn) {
+  
+    try {
+      conn.setAutoCommit(false);
+      // Prepare a statement to execute the SQL Queries.
+      try (Statement statement = conn.createStatement() ) {
+      // Select 20 rows from the CUSTOMERS table from SH schema. 
+      try (ResultSet resultSet = statement.executeQuery ("select CUST_ID,"  
+          +  "CUST_FIRST_NAME, CUST_LAST_NAME, CUST_CITY, CUST_CREDIT_LIMIT "
+          +  "FROM SH.CUSTOMERS WHERE ROWNUM < 20 order by CUST_ID ") ) {
+        System.out.println("\nCUST_ID" + " " + "CUST_FIRST_NAME" + " " + "CUST_LAST_NAME" 
```

Comment:
This can be a single string, no need to concatenate with space. Alternatively use `String.join()` if you want to keep the column names separate.

Reply:
String.join() is used for a different purpose. I could use String.concat() but, + also does the same thing. I moved the query to a variable for better readability.

Reply:
Pick the one that's more readable

1. String.join(" ", "\nCUST_ID", "CUST_FIRST_NAME", "CUST_LAST_NAME", "CUST_CITY", "CUST_CREDIT_LIMIT");
2. "\nCUST_ID".concat(" ").concat("CUST_FIRST_NAME").concat(" ").concat("CUST_LAST_NAME").concat(" ").concat("CUST_CITY").concat(" ").concat("CUST_CREDIT_LIMIT");
3. "\nCUST_ID" + " " + "CUST_FIRST_NAME" + " " + "CUST_LAST_NAME" + " " + "CUST_CITY" + " " + "CUST_CREDIT_LIMIT";

For me it's 1.

Reply:
What we have is readable and I don't think, this needs a change. 
 String queryStatement = "SELECT CUST_ID, CUST_FIRST_NAME, CUST_LAST_NAME, CUST_CITY," 
      + "CUST_CREDIT_LIMIT FROM SH.CUSTOMERS WHERE ROWNUM < 20 order by CUST_ID";

Reply:
The string you linked is not the one I saw. I'm referring to the header that's printed before every row.

Reply:
Done, please check